bitswap/client/internal/sessioninterestmanager/sessioninterestmanager.go

```diff
@@ -131,7 +131,7 @@ func (sim *SessionInterestManager) FilterSessionInterested(ses uint64, ksets ...
 		// For each key in the list
 		for _, c := range ks {
 			// If there is a session that's interested, add the key to the set
-			if _, ok := sim.wants[c][ses]; ok {
+			if wanted, ok := sim.wants[c][ses]; ok && wanted {
```

Comment:
I think the original code was intentional.
```
// The boolean indicates whether the session still wants the block                                                                                                                                                                                    
// or is just interested in receiving messages about it. 
```
Also, see comments here:
https://github.com/ipfs/boxo/blob/main/bitswap/client/internal/sessioninterestmanager/sessioninterestmanager.go#L153-L156

I think the original code is correct, because it adds the key to the `has` set if the session is interested in the blocks (`wanted==true`) ot just the messages (`wanted==false`). This function is asking if the session has an interest - not specifically if it wants the blocks.

Reply:
Thanks so much!

For my own notes: `FilterSessionInterested` is only called from https://github.com/ipfs/boxo/blob/73057e2145e4f105cf75ce6894ad07c0a37b6c73/bitswap/client/internal/session/session.go#L194 which then calls `sws.Update`. 

sws.Update will then removeWant() for the keys that we were "interested in" when it processes the update as it contains the cids of the blocks received. iirc this also means that the wants for a peer a both tracked in their own ways in sessionInterestManager and SessionWantSender.